snap/info.go

```diff
@@ -1728,6 +1728,35 @@ func SplitSnapComponentInstanceName(name string) (snapInstance, componentName st
 	return snapInstance, componentName
 }
 
+// SplitSnapInstanceAndComponents splits a name of the form
+// <snap_instance>+<comp1>...+<compN>.
+func SplitSnapInstanceAndComponents(name string) (string, []string) {
+	parts := strings.Split(name, "+")
+	return parts[0], parts[1:]
+}
+
+// SnapInstancesAndComponentsFromNames splits a slice of names of the form
+// <snap_instance>+<comp1>...+<compN> into a slice of snap instances and a map
+// from these instances to components.
+func SnapInstancesAndComponentsFromNames(names []string, forInstall bool) ([]string, map[string][]string) {
```

Comment:
Is there a case for this just being a local helper in the `main` package? The use-case seems somewhat niche. If not, it should probably have tests here in the `snap` package.

Reply:
Moved now - I have the suspicion that eventually we might need it in `snap`, but let's move it for the moment. It was tested already by `TestSnapInstancesAndComponentsFromNames`, which I have moved now to `snap` too.